lib/spack/spack/test/install.py

```diff
@@ -53,6 +53,81 @@ def test_install_and_uninstall(install_mockery, mock_fetch, monkeypatch):
         raise
 
 
+def test_pkg_attributes(install_mockery, mock_fetch, monkeypatch):
+    # Get a basic concrete spec for the dummy package.
+    spec = Spec('attributes-foo-app')
+    spec.concretize()
+    assert spec.concrete
+
+    # Get the package
+    pkg = spec.package
+    try:
+        pkg.do_install()
+
+        # Validate the virtuals are provided by the base package
+        foo_spec = spec.normalized()['attributes-foo']
```

Comment:
Is `.normalized()` necessary here? I can't think what it would be achieving.

Reply:
Also we'll likely get rid of `normalized()` entirely some day soon (it's only really a concept within the old concretizer).

Reply:
`.normalized()` was necessary to get to the underlying package spec  since the only direct dependencies are the virtual packages.  For the sake of the test though it doesn't really matter and I can setup the dependencies a bit different and get rid of `.normalize`